core/DataAccess/ArchivingDbAdapter.php

```diff
@@ -31,11 +33,17 @@ class ArchivingDbAdapter
      */
     private $logger;
 
+    /**
+     * @var int
+     */
+    private $maxExecutionTime;
+
     public function __construct($wrapped, Lock $archivingLock = null, LoggerInterface $logger = null)
     {
         $this->wrapped = $wrapped;
         $this->archivingLock = $archivingLock;
         $this->logger = $logger;
+        $this->maxExecutionTime = Config::getInstance()->General['archiving_query_max_execution_time'];
```

Comment:
Do we need to check for malformed input at all?

Reply:
Don't think we are doing that for other config values that much. But will add a casting to float to ensure it's at least a number.